lib/ansible/plugins/lookup/hashi_vault.py

```diff
@@ -185,6 +197,17 @@ def boolean_or_cacert(self, validate_certs, cacert):
         else:
             return False
 
+    def auth_approle(self, **kwargs):
```

Comment:
This doesn't appear to ever be called by anything, is there some kind of introspection going on?

Reply:
Yes,

```python
#to enable a new auth backend, simply add a new 'def auth_<type>' method below.
...
# prefixing with auth_ to limit which methods can be accessed
getattr(self, 'auth_' + self.auth_method)(**kwargs)
```

From #20244

Reply:
:+1: